From a683deabb16e9ba1c577f23d8353e22b3bc7f64f Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 11:40:41 -0400
Subject: [net] ethtool: Improve explanation of the two arrays following struct ethtool_rxfh

Message-id: <1410176446-10440-6-git-send-email-ivecera@redhat.com>
Patchwork-id: 91653
O-Subject: [RHEL7.1 PATCH 05/10] ethtool: Improve explanation of the two arrays following struct ethtool_rxfh
Bugzilla: 1138823
RH-Acked-by: Michal Schmidt <mschmidt@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: #1138823

Upstream commit(s):
commit 38c891a49dec43dbb1575cc40d10dbd49c4961ab
Author: Ben Hutchings <ben@decadent.org.uk>
Date:   Thu May 15 01:07:16 2014 +0100

    ethtool: Improve explanation of the two arrays following struct ethtool_rxfh

    The use of two variable-length arrays is unusual so deserves a bit
    more explanation.

    Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/uapi/linux/ethtool.h b/include/uapi/linux/ethtool.h
index 8c47969..c51382e 100644
--- a/include/uapi/linux/ethtool.h
+++ b/include/uapi/linux/ethtool.h
@@ -633,16 +633,17 @@ struct ethtool_rxfh_indir {
  * struct ethtool_rxfh - command to get/set RX flow hash indir or/and hash key.
  * @cmd: Specific command number - %ETHTOOL_GRSSH or %ETHTOOL_SRSSH
  * @rss_context: RSS context identifier.
- * @indir_size: On entry, the array size of the user buffer, which may be zero,
- *  or (for %ETHTOOL_SRSSH), %ETH_RXFH_INDIR_NO_CHANGE.
- *  On return from %ETHTOOL_GRSSH, the array size of the hardware
- *  indirection table.
- * @key_size: On entry, the array size of the user buffer in bytes,
- *  which may be zero.
- *  On return from %ETHTOOL_GRSSH, the size of the RSS hash key.
+ * @indir_size: On entry, the array size of the user buffer for the
+ * indirection table, which may be zero, or (for %ETHTOOL_SRSSH),
+ * %ETH_RXFH_INDIR_NO_CHANGE.  On return from %ETHTOOL_GRSSH,
+ * the array size of the hardware indirection table.
+ * @key_size: On entry, the array size of the user buffer for the hash key,
+ * which may be zero.  On return from %ETHTOOL_GRSSH, the size of the
+ * hardware hash key.
  * @rsvd: Reserved for future extensions.
  * @rss_config: RX ring/queue index for each hash value i.e., indirection table
- *  of size @indir_size followed by hash key of size @key_size.
+ * of @indir_size __u32 elements, followed by hash key of @key_size
+ * bytes.
  *
  * For %ETHTOOL_GRSSH, a @indir_size and key_size of zero means that only the
  * size should be returned.  For %ETHTOOL_SRSSH, an @indir_size of
-- 
1.7.1